solr/core/src/java/org/apache/solr/rest/RestManager.java

```diff
@@ -76,7 +76,7 @@
   private static class ManagedResourceRegistration {
     String resourceId;
     Class<? extends ManagedResource> implClass;
-    List<ManagedResourceObserver> observers = new ArrayList<>();
+    Set<ManagedResourceObserver> observers = new HashSet<>();
```

Comment:
I propose using LinkedHashSet so that these get initialized in a consistent predictable order.  Ideally this doesn't matter but it can help in debugging and in case some custom components are ordering dependent.

Reply:
I can change this, i found no test case which will test ordering explicitly.